catalog/internals/scripts/connect-stack.md

```diff
@@ -0,0 +1,142 @@
+
+This script relies on the quilt3 CLI and does not require additional environment variables.
+
+### Examples
+
+```bash
+# Use quilt3 config (runs 'quilt3 config' to get URL)
+node connect-stack.js
+
+# Use specific stack URL (overrides quilt3 config)
+node connect-stack.js --stack https://my-stack.example.com
+
+# Skip authentication
+node connect-stack.js --no-auth
+
+# Combined options
+node connect-stack.js --stack https://my-stack.example.com --no-auth
+```
+
+### How It Works
+
+1. **Get Config**: The script runs the `quilt3 config` CLI command to retrieve the active stack URL
+
+2. **Resolve Stack URL**: Uses the CLI output or honors the `--stack` override when provided
+
+3. **Authenticate** (optional):
+   - Launches Chrome via `chrome-launcher`
+   - Navigates to the stack's login page
+   - Waits for user to complete authentication
+   - Extracts authentication tokens from cookies/localStorage
+
+4. **Fetch Stack Info**: Attempts to fetch the stack's configuration from `{stackUrl}/config.json`
+
+5. **Generate Config**: Creates a configuration object that matches the `ConfigJson` TypeScript interface
+
+6. **Write Config File**: Writes the configuration as a JavaScript file to `static-dev/config.js` that sets `window.QUILT_CATALOG_CONFIG`
+
+### Output
+
+The script generates a `config.js` file in the `static-dev` directory that looks like:
+
+```javascript
+// Auto-generated by connect-stack.js
+// Do not edit this file directly
+window.QUILT_CATALOG_CONFIG = {
+  "region": "us-east-1",
+  "mode": "LOCAL",
+  "alwaysRequiresAuth": false,
+  "serviceBucket": "your-service-bucket",
+  "apiGatewayEndpoint": "https://your-stack.example.com/api",
+  "registryUrl": "https://your-stack.example.com",
+  "s3Proxy": "https://your-stack.example.com/proxy",
+  // ... other configuration options
+};
+```
+
+This file is automatically served by the webpack dev server and loaded by the catalog application.
+
+### Dependencies
+
+- `chrome-launcher`: For driving a local Chrome instance
+- `chrome-remote-interface`: For interacting with Chrome DevTools Protocol
+
+### Error Handling
+
+The script includes comprehensive error handling for:
+
+- Missing stack URL configuration (via `quilt3 config` or `--stack`)
+- Invalid URLs
+- Network errors when fetching stack info
+- Authentication timeouts
+- Chrome launch or DevTools connectivity issues
+
+### Integration with Webpack
+
+The generated `config.js` file is placed in the `static-dev` directory, which is served by the webpack development server. The main `index.html` includes this file with:
+
+```html
+<script defer src="/config.js"></script>
+```
+
+The `Config.ts` module then reads the configuration from `window.QUILT_CATALOG_CONFIG`.
+
+### NPM Scripts
+
+The following npm scripts are available:
+
+- `npm run connect:stack`: Run the connection script
+- `npm run start:connected`: Connect to stack and start dev server
+- `npm run start`: Start dev server (expects config to already be generated)
```

Comment:
`npm start` is used for a different task already, and it is not defined in this PR.

I mean, it starts catalog dev server, not the "dev server" from this PR.